### PR TITLE
Removed EngineCtl singleton to improve testability and isolation

### PR DESCRIPTION
**Pull request by reinkrul**
Replaced with System type which isn't a singleton.

### CODE REVIEW

File: core/config.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ const (
 	// GlobalServerMode is used for starting the application in server mode, running as Nuts node.
 	GlobalServerMode string = "server"
 	// GlobalCLIMode is used for starting the application in CLI mode, meaning it's used as CLI client administering
-	// for a remote Nuts node. Engines will start in client mode when this mode is specified.
+	// for a remote Nuts node. engines will start in client mode when this mode is specified.
```
Comment (woutslakhorst) [inline core/config.go:90]: no comment

File: core/engine.go [hunk 2]
```diff
@@ -27,13 +27,85 @@ import (
...
+type System struct {
+	// engines is the slice of all registered engines
+	engines []*Engine
+}
+
+// Diagnostics returns the compound diagnostics for all engines.
+func (system *System) Diagnostics() []DiagnosticResult {
+	result := make([]DiagnosticResult, 0)
+	system.VisitEngines(func(engine *Engine) {
+		if engine.Diagnosable != nil {
```
Comment (woutslakhorst) [inline core/engine.go:45]: we could also go to a system where you check if an engine can be cast to a Diagnosable? Removing the need to register it in the engine struct?

**Review by woutslakhorst — COMMENTED** [review]
See other comment: what if we use interfaces: diagnosable, runnable, configurable and check if the engine supports that. Then remove func references from struct?

**Comment by reinkrul** [comment]
> See other comment: what if we use interfaces: diagnosable, runnable, configurable and check if the engine supports that. Then remove func references from struct?

I started to rework exactly this; make Engine an interface which might implement these interfaces you mentioned. Got stuck with the 'engine instances'; since each engine is essentially composed of core functionality ("instance", e.g. `VDR` or `KeyStore`) and something we call an "engine" which attaches to it to configure and manage its lifecycle.

I'd like to merge this first and then we can see how to rework the `Engine` struct.

File: core/engine.go (hunk 2 above)
Comment (reinkrul) [reply core/engine.go:45]: See comment below

**Review by woutslakhorst — APPROVED** [review]
Please track engine struct -> interface refactor with an issue

**Comment by reinkrul** [comment]
https://github.com/nuts-foundation/nuts-node/issues/29

**Pull request merged**